lib/RGBot.js

```diff
@@ -400,39 +769,142 @@ const RGBot = class {
+                    } catch (ex) {
+                        this.#logDebug(`Failed to get near enough to entity to attack`);
+                    }
+                }
+            } catch (ex) {
+                this.#logWarn(`Error trying to attack entity: ${entity.username || entity.name}`, ex)
             }
         }
+        return false
+    }
+
+    /**
+     * <i><b>Experimental</b></i>
+     *
+     * This uses lastAttackTime,lastAttackItem variables to manage weapon attack cool-downs.
+     * This assumes that the weapon you just attacked with needs to cool-down before you can attack again,
+     * even if that next attack is with a different weapon.
+     *
+     * Note: This currently only handles melee weapons
+     *
+     * @returns {Promise<void>}
+     */
+    async waitForWeaponCoolDown() {
+        const attackItem = this.lastAttackItem
+        let ticksSinceAttack = (Date.now()-this.lastAttackTime)/(1000*0.05)
+        let weaponTicks = 5 // wait 0.25 sec for attacking with anything else in your hand
+        if (attackItem) {
+            let itemName = attackItem.name.toLowerCase()
+            if (itemName.includes("sword")) {
+                // wait 0.6 second between attacks for swords for 100% damage
+                weaponTicks = 12
+            } else if (itemName.includes("trident")) {
+                weaponTicks = 18 // wait 0.9 second between attacks for tridents for 100% damage
+            } else if (itemName.endsWith("_axe")) {
+                //axes speed actually varies by type...
+                if (itemName.startsWith("wooden") || itemName.startsWith("stone")) {
+                    weaponTicks = 25 // wait 1.25 second between attacks for wooden/stone axe for 100% damage
+                } else if (itemName.startsWith("iron")) {
+                    weaponTicks = 22 // wait 1.1 second between attacks for iron axe for 100% damage
+                } else {
+                    weaponTicks = 20 // wait 1 second between attacks for gold/diamond/netherite axe for 100% damage
+                }
+            } else if (itemName.endsWith("_pickaxe")) {
+                weaponTicks = 17 // wait 0.85 second between attacks for pickaxe for 100% damage
+            }
+        }
+        // MUST be a whole number
+        let ticksToWait = Math.round(Math.max(0, weaponTicks-ticksSinceAttack))
+        if (ticksToWait > 0){
+            this.#logDebug(`Waiting ${ticksToWait} ticks for weapon cool down...`)
+            await this.wait(ticksToWait)
+        }
+    }
+
+    /**
+     * Moves the bot to at least the specified distance away from the position indicated.
+     * This draws a vector on the XZ plane from the position through the player and finds
+     * the point at the specified distance.  The bot will move to that point unless it is
+     * already further away than the distance.
+     *
+     * @param {Vec3} position
+     * @param {number} distance
+     * @returns {Promise<boolean>}
```

Comment:
Clarify what this boolean means? That it means is it successfully that distance or further.